sqldelight-gradle-plugin/src/test/fixtures/well-formed-selects/expected/com/sample/HockeyPlayerModel.java

```diff
@@ -716,5 +673,143 @@ public Calendar map(Cursor cursor) {
     public Mapper<T> inner_joinMapper() {
       return new Mapper<T>(this);
     }
+
+    private final class For_teamQuery extends SqlDelightQuery {
+      private final long _id;
+
+      For_teamQuery(long _id) {
+        super("SELECT *\n"
+            + "FROM hockey_player\n"
+            + "JOIN team ON hockey_player.team = team._id\n"
+            + "WHERE team._id = ?1",
+            new TableSet("hockey_player", "team"));
+
+        this._id = _id;
+      }
+
+      @Override
+      public void bindTo(SupportSQLiteProgram program) {
+        program.bindLong(1, _id);
+      }
+    }
+
+    private final class Question_marks_everywhereQuery extends SqlDelightQuery {
+      @Nullable
+      private final Object arg1;
+
+      @Nullable
+      private final Object arg2;
+
+      @Nullable
+      private final Object arg3;
+
+      private final long arg4;
+
+      @Nullable
+      private final Object arg5;
+
+      private final long arg6;
+
+      Question_marks_everywhereQuery(@Nullable Object arg1, @Nullable Object arg2,
+          @Nullable Object arg3, long arg4, @Nullable Object arg5, long arg6) {
+        super("SELECT _id\n"
+            + "FROM hockey_player\n"
+            + "WHERE ?1 = ?2\n"
+            + "GROUP BY ?3 HAVING ?4\n"
```

Comment:
Why is 4 a long?

Reply:
Shouldn't 1 and 5 be a string?

Reply:
HAVING is resolved as a boolean by sqlite - ie 1 or 0.

You can order by any type

idk why 1 would be a string?

Reply:
It's likely to be a column, and I'd be tempted to require a column on the left and require arbitrary values only on the right side of a binary operator

Reply:
for all binary operators seems like overboard but for the most part i agree

`WHERE some_column LIKE "%" || ?1 || "%"` -> the pipes would fail

also do string column names even work? `WHERE 'some_column' = ?` I've never tried

Reply:
Ah, right. Probably not.